styles/all/template/js/collapsiblecategories.js

```diff
@@ -0,0 +1,58 @@
+(function ($) {  // Avoid conflicts with other libraries
+
+	'use strict';
+
+	$(function () {
+		$('a.category[data-id] + .forabg').each(function () {
+			var $this = $(this),
+				id = $this.prev().attr('data-id'),
+				hidden = $this.prev().attr('data-hidden'),
+				$header = $this.find('li.header'),
+				$content = $this.find('.topiclist.forums');
+
+			if (!$header.length || !$content.length) {
+				return;
+			}
+
+			// Add button
+			$header.append('<a class="collapse-btn collapse-' + ((hidden) ? 'show' : 'hide') + '" data-id="' + id + '" data-ajax="phpbb_collapse"></a>');
```

Comment:
An alternative to hardcoding all this HTML, which I think would be better, and more readable, is this:
```javascript
var $button = $('<a>')
	.addClass('collapse-btn collapse-' + ((hidden) ? 'show' : 'hide'))
	.attr({
		href: collapsible_categories_url + id,
		'data-ajax': 'phpbb_collapse',
		'data-overlay': true
	});
$header.append($button);
```
Also we need an href for the data-ajax to work. The var `collapsible_categories_url` would be set back in a template event like:
```
<script>
	var collapsible_categories_url = '{{ ROOT_PATH }}collapse/';
</script>
```
Note that I'm using TWIG syntax there :)